components/__tests__/UnitInput.test.ts

```diff
@@ -0,0 +1,163 @@
+import { mount } from '@vue/test-utils';
+// @ts-ignore
+import UnitInput from '@/components/form/UnitInput.vue';
+// @ts-ignore
+import LabeledInput from '@/components/form/LabeledInput.vue';
+import { UNITS } from '../../utils/units';
+
+describe('unitinput', () => {
+  test('renders', () => {
+    const wrapper = mount(UnitInput, { propsData: { value: 1 } });
+
+    expect(wrapper.isVueInstance).toBeTruthy();
+  });
+
+  test('emits input event when input value changes', () => {
+    const wrapper = mount(UnitInput, { propsData: { value: 1 } });
+    const inputWrapper = wrapper.find('input');
+
+    inputWrapper.setValue(2);
+    inputWrapper.setValue(4);
+
+    expect((wrapper.emitted('input') || []).length).toBe(2);
+  });
+
+  test("input is disabled when mode='view'", () => {
+    const wrapper = mount(UnitInput, { propsData: { value: 1, mode: 'view' } });
+    const inputElement = wrapper.find('input').element as HTMLInputElement;
+
+    expect(inputElement.disabled).toBeTruthy();
+  });
+
+  test('Input strings with any si modifier are parsed', async() => {
+    const wrapper = mount(UnitInput, { propsData: { value: '1000m' } });
+    const labeledInputWrapper = wrapper.findComponent(LabeledInput);
+
+    expect(labeledInputWrapper.props('value')).toBe('1');
+
+    wrapper.setProps({ value: '2G' });
+    await wrapper.vm.$nextTick();
+
+    expect(labeledInputWrapper.props('value')).toBe('2000000000');
+
+    wrapper.setProps({ value: '3' });
+    await wrapper.vm.$nextTick();
+
+    expect(labeledInputWrapper.props('value')).toBe('3');
+  });
+
+  test('Prop inputExponent determines unit prefix', () => {
+    const wrapper = mount(UnitInput, { propsData: { value: 1, inputExponent: 2 } });
+    const expectedUnits = `${ UNITS[2] }B`;
+
+    const suffixDiv = wrapper.find('.addon');
+
+    expect((suffixDiv.element.textContent || '').trim() === expectedUnits).toBeTruthy();
+  });
+
+  test('Prop outputModifier toggles inclusion of si modifier in output', async() => {
+    const wrapper = mount(UnitInput, {
+      propsData: {
+        value: 1, inputExponent: 3, outputModifier: true
+      }
+    });
+    const inputWrapper = wrapper.find('input');
+
+    inputWrapper.setValue(2);
+    let inputEventValue = wrapper.emitted('input')[0][0];
```

Comment:
These returned values in Typescript may be `undefined` and in a correct configuration they will return error.

This is an example I created to extract the concept:

Without linting, only reassignment is flagged:
![Screenshot from 2022-02-16 11-16-56](https://user-images.githubusercontent.com/5009481/154258667-d0f7576d-f7f7-4879-9b80-a5fe4a5ec10d.png)

With linting, possible issues are highlighted:
![Screenshot from 2022-02-16 11-18-02](https://user-images.githubusercontent.com/5009481/154258786-511192ef-b23f-4457-8a02-02b8334b8734.png)

This is convenient because you may have typo or erroneously accessing to an object or have a not updated type. In the example we try to access `['random key']` which is a clear error.